### PR TITLE
Upgrade vendored Brotli dependency to v1.1.0

### LINKED ISSUE #106887: Upgrade native brotli to version 1.0.10
We're using 1.0.9, released on August 2020: https://github.com/dotnet/runtime/blob/f402418aaed508c1d77e41b942e3978675183bfc/src/native/external/brotli-version.txt#L2

The latest is 1.0.10, released on August 2023: https://github.com/google/brotli/releases/tag/v1.1.0

### PR DESCRIPTION
**Pull request by jkoritzinsky**
Fixes #106887

### CODE REVIEW

File: src/native/libs/System.IO.Compression.Native/CMakeLists.txt [hunk 1]
```diff
@@ -104,6 +98,10 @@ if (CLR_CMAKE_TARGET_UNIX OR CLR_CMAKE_TARGET_BROWSER OR CLR_CMAKE_TARGET_WASI)
 
     set_target_properties(System.IO.Compression.Native-Static PROPERTIES OUTPUT_NAME System.IO.Compression.Native CLEAN_DIRECT_OUTPUT 1)
 else ()
+
+    # Disable C4334 warning for Zlib and Brotli
+    add_compile_options($<$<COMPILE_LANG_AND_ID:C,MSVC>:/wd4334>)
```
Comment (carlossanlop) [inline src/native/libs/System.IO.Compression.Native/CMakeLists.txt:103]: For anyone interested, I fixed 4334 upstream for zlib-ng, and two other errors: https://github.com/zlib-ng/zlib-ng/pull/1762

If you're seeing the error in brotli, maybe consider submitting an upstream PR for them too?

**Review by carlossanlop — COMMENTED** [review]
The changes look good. Can you please run a few additional azp runs? For example, runtime-community, runtime-extra-platforms, and the nativeaot and wasm ones

File: src/native/libs/System.IO.Compression.Native/CMakeLists.txt (hunk 1 above)
Comment (jkotas) [reply src/native/libs/System.IO.Compression.Native/CMakeLists.txt:103]: Is 4334 on the must-fix SDL list?

Comment (jkoritzinsky) [reply src/native/libs/System.IO.Compression.Native/CMakeLists.txt:103]: Yes, it is on the must fix list. It's a level 3 warning and all level 3 warnings are on the must-fix list.

I'll prep a fix and push it here.

Comment (jkoritzinsky) [reply src/native/libs/System.IO.Compression.Native/CMakeLists.txt]: Pushed a fix

**Review by jkotas — APPROVED** [review]

File: src/native/external/brotli-version.txt [hunk 2]
```diff
@@ -1,2 +1,6 @@
-1.0.9
-https://github.com/google/brotli/releases/tag/v1.0.9
+1.1.0
+https://github.com/google/brotli/releases/tag/v1.1.0
+
+Copy the c/dec, c/enc, c/common, and c/include folders into the root and remove all other files.
```
Comment (am11) [inline src/native/external/brotli-version.txt:4]: We should probably create a script to do this, if we are going to selectively copy stuff. Previously, the idea was to keep these external dependencies as-is, so it's easy to compare/review during the updates.

Comment (jkoritzinsky) [reply src/native/external/brotli-version.txt:4]: I'm planning (and have a branch locally that I need to update) that will bring us in alignment with how we handle other dependencies.

**Pull request merged**